host/CMakeLists.txt

```diff
@@ -282,10 +282,17 @@ if (OE_SGX)
       PLATFORM_SDK_ONLY_SRC
       sgx/linux/aep.S
       sgx/enter.c
+      sgx/exception_sim.c
       sgx/linux/exception.c
       sgx/linux/sgxioctl.c
       sgx/linux/switchless.c
       sgx/linux/xstate.c)
+
+    # Functions in sgx/exception_sim.c will change the fs register and may trigger
+    # stack check fail, thus it is necessary to turn off stack check.
+    set_source_files_properties(sgx/exception_sim.c
```

Comment:
For my understanding, can you elaborate on the mechanism of failure? Is it because stack protection injects a cookie at the exception handler address, and so sees the enclave FS before we have the opportunity to fix it back up to the host value inside the exception handler itself?